<a href="https://colab.research.google.com/github/khatriprajwol/Data-Visualization-/blob/main/SQL%20with%20BigQuery.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## SQL with BigQuery

In this project, I will be using the basic knowledge which I have learned in the class. 

In [64]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


This code will is used to authorize the permission to use the data which included several process. 

In [65]:
%%bigquery --project data-base-1-338717
SELECT 
  COUNT(*) as total_rows
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_rows
0     1424786

Wow! It looks like there is total of 1424786 rows in the data set which is a lot. 

In [66]:
%%bigquery --project data-base-1-338717
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
LIMIT 10

trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19
5  9900286143  ...               17
6  9900286171  ...                6
7  9900286214  ...                0
8  9900286540  ...               12
9    13575843  ...               47

[10 rows x 9 columns]

This input gives me only 10 rows even though it has total of 1424786 rows. That is really helpful to over view in a small table which beautifully presents the information. 

In [67]:
%%bigquery --project data-base-1-338717 df
SELECT 
  duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

In [68]:
df

duration_minutes
0                      47
1                      59
2                      58
3                      82
4                      76
...                   ...
1424781                43
1424782                43
1424783                43
1424784                43
1424785                43

[1424786 rows x 1 columns]

In [69]:
%%bigquery --project data-base-1-338717
SELECT AVG(duration_minutes) as duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as table

duration_minutes
0         30.870428

30.870428 is the average *duration_minutes* for the trip. 

In [70]:
%%bigquery --project data-base-1-338717
SELECT MAX(duration_minutes) as duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips` as table

duration_minutes
0             34238

34238 is the maximum *duration_minutes* time for the trip.:

Hence, with out using `panda` as a datafram we can do this in a simple way. However, using `panda` is also a good idea to do it in a different way.

In [76]:
import pandas as pd

groupeddf = df.groupby('duration_minutes')

In [78]:
print(df['duration_minutes'].max())

34238


In [80]:
print(df['duration_minutes'].mean())

30.870427558945693


It is just a different process but the output is same.